rust/otap-dataflow/crates/otap/src/otap_receiver.rs

```diff
@@ -840,86 +775,153 @@ mod tests {
         let pipeline_ctx =
             controller_ctx.pipeline_context_with("grp".into(), "pipeline".into(), 0, 0);
 
-        // Test with custom max_concurrent_requests, max_concurrent_requests defaults to 1000
+        // Test with custom max_concurrent_requests, max_concurrent_requests defaults to 0
         let config_with_max_concurrent_requests = json!({
             "listening_addr": "127.0.0.1:4317",
-            "response_stream_channel_size": 100,
             "max_concurrent_requests": 5000
         });
         let receiver =
             OTAPReceiver::from_config(pipeline_ctx.clone(), &config_with_max_concurrent_requests)
                 .unwrap();
-        assert_eq!(receiver.config.listening_addr.to_string(), "127.0.0.1:4317");
-        assert_eq!(receiver.config.response_stream_channel_size, 100);
-        assert_eq!(receiver.config.max_concurrent_requests, 5000);
-        assert!(!receiver.config.wait_for_result);
-        assert!(receiver.config.compression_method.is_none());
-        assert!(receiver.config.timeout.is_none());
-
-        // Test with minimal required fields, max_concurrent_requests defaults to 1000, wait_for_result defaults to false
+        assert_eq!(
+            receiver.config.grpc.listening_addr.to_string(),
+            "127.0.0.1:4317"
+        );
+        assert_eq!(receiver.config.grpc.max_concurrent_requests, 5000);
```

Comment:
For one of these tests, we should also verify that `tune_max_concurrent_requests` method works correctly.